secboot/luks2/cryptsetup.go

```diff
@@ -142,77 +121,36 @@ func AddKey(devicePath string, existingKey, key []byte, options *AddKeyOptions)
 		options = &AddKeyOptions{Slot: AnySlot}
 	}
 
-	fifoPath, cleanupFifo, err := mkFifo()
-	if err != nil {
-		return xerrors.Errorf("cannot create FIFO for passing existing key to cryptsetup: %w", err)
-	}
-	defer cleanupFifo()
-
 	args := []string{
 		// add a new key
 		"luksAddKey",
 		// LUKS2 only
 		"--type", "luks2",
-		// read existing key from named pipe
-		"--key-file", fifoPath}
+		// read existing key from stdin, specifying size so no new line
+		// separator is needed (as the existing key can contain them)
+		"--key-file", "-",
+		"--keyfile-size", strconv.Itoa(len(existingKey)),
```

Comment:
The man page says:

```
If the name given is "-", then the passphrase will be read from stdin. In this case, reading will not stop at newline characters.
```

So it looks to me that `--keyfile-size` is not necessary.

I have digged into the git history, and even in the initial commit in 2008, it behaved like that. That means since 1.0.7.

Focal seems to have 2.2.2. Do you remember where you saw the issue?

Reply:
OK, so now, I understand. I think the comment was confusing. We should say that we provide 2 keys on stdin and `keyfile-size` is to split them.

Reply:
I've expanded now the comments so this is more clear now.